src/components/types.ts

```diff
@@ -30,11 +29,19 @@ export interface PosterProps {
   domain?: string;
 }
 
-export interface VideoProps extends Omit<DefaultPlayerProps, 'src' | 'poster'> {
+type DefaultPlayerProps = Omit<MediaProps, 'ref'> & PlayerProps;
+
+// Keep in mind the hierarchy is Video(Player(Media))
+
+export type VideoProps<TPlaybackId = string | undefined> = TPlaybackId extends string
+  ? Omit<Partial<MuxVideoProps> & PlayerProps, 'src' | 'poster'> & SuperVideoProps & { playbackId?: TPlaybackId }
+  : Omit<NativeVideoProps & PlayerProps, 'src' | 'poster'> & SuperVideoProps & { playbackId?: undefined };
```

Comment:
this makes it so that when a playbackId is inferred from the src the types adapt to MuxVideoProps, otherwise restricted to native video props. didn't know this was possible since a few days ago, did take some time to figure it out but it's so good!